hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMDBCheckpointServlet.java

```diff
@@ -594,9 +594,21 @@ private void writeFilesToArchive(
     if (completed) {
       // Only create the hard link list for the last tarball.
       if (!hardLinkFiles.isEmpty()) {
-        Path hardLinkFile = createHardLinkList(truncateLength, hardLinkFiles);
-        includeFile(hardLinkFile.toFile(), OmSnapshotManager.OM_HARDLINK_FILE,
-            archiveOutputStream);
+        Path hardLinkFile = null;
+        try {
+          hardLinkFile = createHardLinkList(truncateLength, hardLinkFiles);
+          includeFile(hardLinkFile.toFile(), OmSnapshotManager.OM_HARDLINK_FILE,
+              archiveOutputStream);
+        } finally {
+          if (Objects.nonNull(hardLinkFile)) {
+            try {
+              Files.delete(Objects.requireNonNull(hardLinkFile));
```

Comment:
`Objects.requireNonNull()` is not be required after adding `Objects.nonNull()` check.

Reply:
Definitely, it was a leftover. Fixed.